ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -229,5 +366,35 @@ long getCapacityInBytes() {
     List<String> getLogSegmentsNotInJournal() throws StoreException {
       return validLogSegments;
     }
+
+    @Override
+    void compact(CompactionDetails details) {
+      compactCalled = true;
+      compactCallsCountdown.countDown();
+      if (details == null) {
+        callOrderException = new Exception("Called compact() with null details");
+      } else if (!resumeCompactionCalled) {
+        callOrderException = new Exception("Called compact() before calling maybeResumeCompaction()");
```

Comment:
This is not intended to mimic PROD code in the store. This is intended to check the behavior of the compaction thread which _must_ call `resumeCompaction()` on a store (when it starts up) before calling `compact()` (repeatedly if required). The call order therefore must be checked - which is what this is doing.